### PR TITLE
Add support for running tests against origin-metrics

### PR DESCRIPTION
**Pull request by jcantrill**
cc @danmcp

### CODE REVIEW

**Comment by danmcp** [comment]
@dobbymoodge Want to take a shot at reviewing?

**Comment by dobbymoodge** [comment]
sure

File: lib/vagrant-openshift/action.rb [hunk 1]
```diff
@@ -173,6 +173,26 @@ def self.repo_sync_origin_aggregated_logging(options)
         end
       end
 
+      def self.repo_sync_origin_metrics(options)
+        Vagrant::Action::Builder.new.tap do |b|
+          b.use PrepareSshConfig
+          if options[:source]
+            if options[:clean]
+              b.use(Clean, options)
```
Comment (dobbymoodge) [inline lib/vagrant-openshift/action.rb]: maybe lose the parens for style consistency

File: lib/vagrant-openshift/action.rb [hunk 2]
```diff
@@ -173,6 +173,26 @@ def self.repo_sync_origin_aggregated_logging(options)
...
+              b.use(CloneUpstreamRepositories, options)
+            end
+            b.use(SyncLocalRepository, options)
+            b.use(CheckoutRepositories, options)
+          end
+          # no build support currently
+          # if options[:build]
```
Comment (dobbymoodge) [inline lib/vagrant-openshift/action.rb]: Do we need to keep this comment block?

**Comment by dobbymoodge** [comment]
LGTM so far, had to skim the last bit though. I'll take another look tomorrow first thing.

**Comment by jcantrill** [comment]
@dobbymoodge updated code per comments

File: lib/vagrant-openshift/constants.rb [hunk 3]
```diff
@@ -47,12 +51,19 @@ def self.aggregated_logging_repos
         }
       end
 
+       def self.metrics_repos
```
Comment (dobbymoodge) [inline lib/vagrant-openshift/constants.rb]: I'm not sure about this pattern. Why is this method defined twice? The first definition just calls this one. I'm not sure this works the way you might be expecting it to. Maybe you could get rid of the first definition and change this one to something like
`def self.metrics_repos(env=nil)`
?

**Comment by dobbymoodge** [comment]
@jcantrill @danmcp LGTM except for one nit here: https://github.com/openshift/vagrant-openshift/pull/450#discussion_r65357495

**Comment by jcantrill** [comment]
@danmcp @dobbymoodge Most of this is copy and paste but I removed the code you identified.

**Comment by danmcp** [comment]
@jcantrill Getting a compile error: https://travis-ci.org/openshift/vagrant-openshift/builds/134448925

File: lib/vagrant-openshift/constants.rb [hunk 4]
```diff
@@ -52,7 +52,10 @@ def self.repos_for_name(reponame)
           'origin' => openshift_repos,
           nil => openshift_repos,
           'origin-console' => console_repos,
-          'origin-aggregated-logging' => aggregated_logging_repos
+          'origin-aggregated-logging' => aggregated_logging_repos,
+          'origin-metrics' => {
```
Comment (dobbymoodge) [inline lib/vagrant-openshift/constants.rb]: This isn't a valid hash in the `{...}`; I would replace it with a method that returns a hash, to match the pattern of the rest of the file.

**Comment by jcantrill** [comment]
@dobbymoodge @danmcp yes. I missed the 'key' part of the hash... is it better now?

**Comment by dobbymoodge** [comment]
@jcantrill lgtm

@danmcp lgty?

**Pull request merged**